astrolabe/utils.py

```diff
@@ -301,15 +302,24 @@ def get_logs(admin_client, project, cluster_name):
     data = admin_client.groups[project.id].logCollectionJobs.post(**params).data
     job_id = data['id']
     
-    while True:
-        LOGGER.debug('Poll job %s' % job_id)
+    timer = Timer()
+    timer.start()
+    ok = False
+    # Wait 10 minutes for logs to get collected - guess as to how long
+    # it might take
+    timeout = 600
+    while timer.elapsed < timeout:
+        LOGGER.info(f"Cluster %s: waiting for log collection job {job_id} for %.1f sec" % (cluster_name, timer.elapsed))
         data = admin_client.groups[project.id].logCollectionJobs[job_id].get().data
         if data['status'] == 'IN_PROGRESS':
             sleep(1)
         elif data['status'] == 'SUCCESS':
+            ok = True
             break
         else:
-            raise Exception("Unexpected log collection job status %s" % data['status'])
+            raise Exception("Unexpected log collection job status: %s: %s" % (data['status'], data))
```

Comment:
I think we can raise AstrolabeTestCaseError here https://github.com/mongodb-labs/drivers-atlas-testing/blob/548d08035a663789eaa0c11e82002a22ad8bed2d/astrolabe/exceptions.py#L19